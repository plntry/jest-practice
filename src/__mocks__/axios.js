/* eslint-disable import/no-anonymous-default-export */
export default {
    get: jest.fn(() => Promise.resolve({ data: {} }))
};