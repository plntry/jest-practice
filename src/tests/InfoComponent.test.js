import renderer from "react-test-renderer";
import axios from "axios";
import getGitHubUser from "../services/DataService/index";
import Info from "../components/Info/index";

jest.mock("axios");

describe("List Component", () => {
  it("renders correctly", () => {
    const wrapper = renderer.create(<Info user="yurkovskiy" />);
    const testInstance = wrapper.root;
    // eslint-disable-next-line testing-library/await-async-query
    expect(testInstance.findByProps({ user: "yurkovskiy" })).toBeTruthy();
  });
  it("should return user login", async () => {
    const user = { data: "yurkovskiy" };
    axios.get.mockResolvedValueOnce(user);
    const res = await getGitHubUser(user);
    expect(res.data).toContain("yurkovskiy");
  });
  it("should return error", async () => {
    jest.spyOn(axios, "get").mockRejectedValue(new Error("request error"));
    const user = { data: "Undefined User" };
    const res = getGitHubUser(user);
    await expect(res).rejects.toThrow("request error");
  });
});

// TODO: Your test need to be here instead of fake tests
