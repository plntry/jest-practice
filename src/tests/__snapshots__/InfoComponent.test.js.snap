// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Info Component renders correctly 1`] = `
<div>
  <h3>
    GitHub User Info
  </h3>
  <ul />
</div>
`;

exports[`List Component renders correctly 1`] = `
<div>
  <h3>
    GitHub User Info
  </h3>
  <ul />
</div>
`;
